src/preamble.js

```diff
@@ -678,15 +678,17 @@ function createExportWrapper(name, fixedasm) {
 // we know to ignore exceptions from there since they're handled by callMain directly.
 var abortWrapperDepth = 0;
 
-// A cache for wrappers based on the original function reference so we don't end up
-// creating the same wrappers over and over again
+// A cache for wrappers based on the original function name so we don't end up
+// creating the same wrappers over and over again. For wasmTable functions the
+// keys are the indices (they can't clash with names  since identifiers can't start 
+// with digits)
 var abortWrapperCache = {};
 
 // Creates a wrapper in a closure so that each wrapper gets it's own copy of 'original'
-function makeWrapper(original) {
-  var wrapper = abortWrapperCache[original];
+function makeAbortWrapper(original, name) {
+  var wrapper = abortWrapperCache[name];
```

Comment:
Maybe just remove the cache?    I'm not sure how it can ever help here.. since export names are unique, as are table indexes, so we can never get a cache hit this way can we?

Reply:
Maybe before we could get a cache hit if we export the same function under many different names, but this case should be very rare.

Reply:
I didn't have it before, but I added it after I added wasmTable instrumentation as that gets hit a lot with the same index and I didn't want it to return a new wrapper instance every time. That would wreck the GC :)

I could move it out of `makeWrapper` into the `wasmTable.get` override only and leave the original uncached since that will now never hit the cache anyways (it would before since a lot of functions shared the same invokers).

Reply:
Just thought it was more readable if it was in `makeAbortWrapper` as it always just works even if we add a 3rd instrumentation eventually